master/internal/experiment.go

```diff
@@ -612,6 +617,27 @@ func (e *experiment) setWeight(ctx *actor.Context, weight float64) error {
 	return nil
 }
 
+func (e *experiment) setRP(ctx *actor.Context, msg job.SetResourcePool) error {
+	if sproto.UseK8sRM(ctx.Self().System()) {
+		return fmt.Errorf("kubernetes does not support setting resource pools")
+	}
+
+	if _, err := sproto.GetResourcePool(ctx.Self().System(), msg.ResourcePool, 0, false); err != nil {
+		return fmt.Errorf("invalid resource pool name %s", msg.ResourcePool)
+	}
+
+	resources := e.Config.Resources()
+	resources.SetResourcePool(msg.ResourcePool)
+	e.Config.SetResources(resources)
+
+	if err := e.db.SaveExperimentConfig(e.Experiment); err != nil {
+		return errors.Wrapf(err, "setting experiment %d RP to %s", e.ID, msg.ResourcePool)
+	}
```

Comment:
in this case, the db errs and 
* we respond to the user with the error
* we don't log the error
* we leave the RP change as is in memory
* we don't tell the change RP message to trials

if asha comes along and decides to launch a new trial, it will get launched into the RP indicate by memory, which is different than the config that will appear to users at that point.

maybe we should have the db errs and
* we respond to the user with the error
* we log the error
* we rollback the RP change in memory
* we don't tell the change RP message to trials

or we could just tells the current trials and log that we were unable to persist the change to the DB and it may cause issues in the event of a master crash. but that seems more complicated than worth for the rate of this error. but at least, as is, it's inconsistent to me that future trials would reflect the config in memory but we bail out before telling current trials.